google/cloud/spanner_dbapi/client_side_statement_executor.py

```diff
@@ -22,8 +22,15 @@ def execute(connection, parsed_statement: ParsedStatement):
 
     It is an internal method that can make backwards-incompatible changes.
 
+    :type connection: Connection
```

Comment:
The function signature is not `connection: Connection`. Should we change that so that the function signature matches this documentation?

Reply:
I am not able to import Connection as it leads to circular dependency. Connection -> Cursor -> client_side_statement_executor -> Connection

Cursor also depends on Connection but doesn't import it so as not to have a circular dependency I think. Any idea if we should follow this (dont import) ?

Reply:
Ah, right, import loops.

There is an official workaround for this, though it's slightly clunky: 
 https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

In essence:
- There's a magic pseudo-variable that you can check, that always evaluates to `False` during execution but that type checkers will pretend has a value of `True`.
- Type checkers generally don't execute code.  They just parse it.  So import loops aren't an issue for them.

Also, fun fact:  In Java (and most languages), control-flow constructs such as "if" define new scopes.  Variables declared inside the code-block representing the body of a loop or a conditional, go out of scope when you exit that code-block.  In Python, "if" statements do not define a scope.  Variables (including imports, since an import is just assigning a module object to a variable name) that are for example declared inside an "if", don't go out of scope until the function, class, or module containing the "if" goes out of scope.  (Class and module bodies can contain "if" statements; in Python, imperative logic doesn't need to live inside a function.  For example you could declare a method on a class that only exists `if sys.platform == 'Linux'`.  Imperative logic outside of a function is generally discouraged unless there's a good reason:  If `import` has side effects or becomes computationally expensive, that encourages design patterns that aren't great for maintainability.)